runtime/doc/lsp.txt

```diff
@@ -954,6 +954,20 @@ start_client({config})                                *vim.lsp.start_client()*
                     not be fully initialized. Use `on_init` to do any actions
                     once the client has been initialized.
 
+start_or_attach({config}, {reuse_client})          *vim.lsp.start_or_attach()*
```

Comment:
The name is awkward because it should just be a  parameter on `start_client` :)

Reply:
I renamed it to just `start` because I wasn't happy with the `start_or_attach` either - a `maybe_start_and_attach` would've been more accurate but that's way too mouthful.
Now it has some overlap with `start_client` but assuming that we may make that private at some point in the future I think it's okayish.